src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,151 @@
+    private static StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_validUser() {
+        User validUser = new User();
+        validUser.setAge(20);
+        validUser.setPassword("passwordValid");
+        validUser.setLogin("loginValid");
+
+        registrationService.register(validUser);
+        assertNotNull(storageDao.get("loginValid"));
+        assertEquals(validUser, storageDao.get("loginValid"));
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        User firstUser = new User();
+        firstUser.setLogin("validLogin");
+        firstUser.setPassword("validPassword");
+        firstUser.setAge(20);
+        storageDao.add(firstUser);
+
+        User secondUserWithSameLogin = new User();
+        secondUserWithSameLogin.setLogin("validLogin");
+        secondUserWithSameLogin.setPassword("anotherPassword");
+        secondUserWithSameLogin.setAge(25);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(secondUserWithSameLogin);
+        });
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User userWithShortPassword = new User();
+        userWithShortPassword.setLogin("validLogin");
+        userWithShortPassword.setPassword("short");
+        userWithShortPassword.setAge(20);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithShortPassword);
+        });
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        User userWithNullPassword = new User();
+        userWithNullPassword.setLogin("validLogin");
+        userWithNullPassword.setPassword(null);
+        userWithNullPassword.setAge(20);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithNullPassword);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User userWithShortLogin = new User();
+        userWithShortLogin.setLogin("log");
+        userWithShortLogin.setPassword("validPassword");
+        userWithShortLogin.setAge(20);
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithShortLogin);
+        });
+    }
+
+    @Test
+    void register_minLengthLogin_ok() {
+        User userWithMinLengthLogin = new User();
+
+        String minLengthLogin = "MinLog";
+        userWithMinLengthLogin.setLogin(minLengthLogin);
+        userWithMinLengthLogin.setPassword("validPassword");
+        userWithMinLengthLogin.setAge(20);
+
+        registrationService.register(userWithMinLengthLogin);
+        assertEquals(userWithMinLengthLogin, storageDao.get(minLengthLogin));
+
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(userWithMinLengthLogin);
+        });
```

Comment:
The exception type should be specific. Instead of `RuntimeException`, use the specific exception that the `register` method is expected to throw.